src/app.ts

```diff
@@ -120,22 +120,22 @@ const cachingMiddleware = apicache.options({
 }).middleware;
 
 export interface IAppFactoryParameters {
-  env: NodeEnvironment;
-  appInsightsClient?: appInsights.TelemetryClient;
-  allowNotifyIPSourceRange: readonly CIDR[];
-  allowPagoPAIPSourceRange: readonly CIDR[];
-  allowMyPortalIPSourceRange: readonly CIDR[];
-  allowBPDIPSourceRange: readonly CIDR[];
-  allowSessionHandleIPSourceRange: readonly CIDR[];
-  authenticationBasePath: string;
-  APIBasePath: string;
-  BonusAPIBasePath: string;
-  PagoPABasePath: string;
-  MyPortalBasePath: string;
-  BPDBasePath: string;
+  readonly env: NodeEnvironment;
+  readonly appInsightsClient?: appInsights.TelemetryClient;
+  readonly allowNotifyIPSourceRange: ReadonlyArray<CIDR>;
+  readonly allowPagoPAIPSourceRange: ReadonlyArray<CIDR>;
+  readonly allowMyPortalIPSourceRange: ReadonlyArray<CIDR>;
+  readonly allowBPDIPSourceRange: ReadonlyArray<CIDR>;
+  readonly allowSessionHandleIPSourceRange: ReadonlyArray<CIDR>;
```

Comment:
No, the `readonly` keyword makes the prop like a `const` so it can't be reassigned (es. `myVar.allowNotifyIPSourceRange = myValue` is forbidden). `ReadonlyArray` instead make the array immutable. They work on different layers.